src/ripple/app/tx/impl/PayChan.cpp

```diff
@@ -130,23 +132,34 @@ closeChannel (
     uint256 const& key,
     beast::Journal j)
 {
-    AccountID const src = (*slep)[sfAccount];
-    // Remove PayChan from owner directory
+    STArray const& chanMembers (slep->getFieldArray (sfChannelMembers));
+
+    // Remove PayChan from owner directories
+    for (auto const member : chanMembers)
     {
-        auto const page = (*slep)[sfOwnerNode];
-        TER const ter = dirDelete (view, true, page, keylet::ownerDir (src).key,
+        auto const owner = member[sfAccount];
+        auto const page = member[sfOwnerNode];
+        TER const ter = dirDelete (view, true, page, keylet::ownerDir (owner).key,
             key, false, page == 0, j);
         if (!isTesSuccess (ter))
             return ter;
     }
 
-    // Transfer amount back to owner, decrement owner count
-    auto const sle = view.peek (keylet::account (src));
-    assert ((*slep)[sfAmount] >= (*slep)[sfBalance]);
-    (*sle)[sfBalance] =
-        (*sle)[sfBalance] + (*slep)[sfAmount] - (*slep)[sfBalance];
-    (*sle)[sfOwnerCount] = (*sle)[sfOwnerCount] - 1;
-    view.update (sle);
+    auto const sle0 = view.peek (keylet::account (chanMembers[0][sfAccount]));
+    auto const sle1 = view.peek (keylet::account (chanMembers[1][sfAccount]));
+
+    (*sle0)[sfOwnerCount] = (*sle0)[sfOwnerCount] - 1;
+
+    // Settle channel claims
+    auto diff = chanMembers[0][sfAmount] - chanMembers[1][sfAmount];
```

Comment:
In unidirectional channels, `Amount` was the amount of channel funding and `Balance` was the amount paid. It looks like those definitions have been swapped for bidirectional channels.